src/app/core/utils/ComponentUtils.ts

```diff
@@ -321,17 +321,23 @@ export function GetCameraFit(camera: Camera, objs: CullableObject[], padding: nu
     if (objs.length === 0)
         return [V(), 1];
 
+    const MarginSize = V(camera.getMargin().left - camera.getMargin().right, camera.getMargin().top - camera.getMargin().bottom);
+
     const bbox = CircuitBoundingBox(objs);
-    const finalPos = bbox.getCenter();
+    const newPos = bbox.getCenter();
 
-    const screenSize = camera.getCenter().scale(2); // Bottom right corner of screen
+    const screenSize = camera.getSize().sub(MarginSize); // Bottom right corner of screen
     const worldSize = camera.getWorldPos(screenSize).sub(camera.getWorldPos(V(0,0))); // World size of camera view
 
     // Determine which bbox dimension will limit zoom level
     const xRatio = bbox.getWidth()/worldSize.x;
     const yRatio = bbox.getHeight()/worldSize.y;
     const finalZoom = camera.getZoom()*Math.max(xRatio, yRatio)*padding;
 
+    // Only subtract off 0.5 of the margin offset since currently it's centered on the margin'd
+    //  screen size so only half of the margin on the top/left need to be contributed
+    const finalPos = newPos.sub(MarginSize.scale(0.5 * finalZoom));
```

Comment:
Did you try this with a different margin? like with something on the right / top / bottom? 
I feel like the math here is still slightly wrong because I think you need more than just `MarginSize`, specifically I think you need to be `.sub`ing the vector `V(camera.getMargin().left, camera.getMargin().bottom)` for calculation of `finalPos` and using `MarginSize` only for the first `.sub` for calculation of `screenSize`. Or vice-versa. Or maybe it's somehow fine but I'm skeptical.

Reply:
So I tried it out with the bottom/top margin and yea, the code here doesn't work on it. The fix that seems to work is the vice versa from your suggestion.